packages/data-store/src/utils/digitalCredential/MappingUtils.ts

```diff
@@ -0,0 +1,112 @@
+import { AddDigitalCredentialArgs } from '../../types/digitalCredential/IAbstractDigitalCredentialStore'
+import { DigitalCredentialEntity } from '../../entities/digitalCredential/DigitalCredentialEntity'
+import {
+  CredentialMapper,
+  decodeSdJwtVc,
+  DocumentFormat,
+  IVerifiableCredential,
+  IVerifiablePresentation,
+  SdJwtDecodedVerifiableCredentialPayload,
+} from '@sphereon/ssi-types'
+import {
+  CredentialDocumentFormat,
+  CredentialType,
+  DigitalCredential,
+  NonPersistedDigitalCredential,
+} from '../../types/digitalCredential/digitalCredential'
+import { computeEntryHash } from '@veramo/utils'
+import { createHash } from 'crypto'
+
+function determineCredentialType(raw: string): CredentialType {
+  if (CredentialMapper.hasProof(raw)) {
```

Comment:
Hmzz, did you check this will work. Because you are now passing in a string, which is only a string for the persistence layer. So even a JSON-LD credential is a string at this point. The CredentialMapper of course can work with strings, but it might assume we are talking about JWTs when it is a string. In other words, probably at this point you should still be accepting the original/raw credential

Reply:
yes I have many test cases for this. but because of your other [comment](https://github.com/Sphereon-Opensource/SSI-SDK/pull/165#discussion_r1506800014) I'll pass in the parsed version to the CredentialMapper. that way I don't need to have additional checks for object as string vs jwt/sd_jwt strings

Reply:
changed this to pass in only a jwt/sd_jwt or json document to the CredentialMapper functions

Reply:
removed and added it to the callback params